actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -104,7 +104,7 @@ def request_http_basic_authentication(realm = "Application", message = nil)
 
       def authenticate(request, &login_procedure)
         if has_basic_credentials?(request)
-          login_procedure.call(*user_name_and_password(request))
+          yield(*user_name_and_password(request))
         end
       end
```

Comment:
I noticed a few methods where the name of the block argument ends up unused after these changes. Should these be removed?

Reply:
I think yes, will do that.